actions/tag/calculate-semver/entrypoint/main.go

```diff
@@ -81,17 +83,36 @@ func main() {
 		fail(fmt.Errorf("failed to get repo: unexpected response: %s", dump))
 	}
 
-	prevVersion, err := getLatestVersion(ghClient, config)
+	fmt.Println("Getting the latest release version on the repository")
+	latestReleaseVersion, err := getLatestVersion(ghClient, config)
 	if err != nil {
 		fail(err)
 	}
 
-	// there are no releases on the repo
-	if prevVersion == nil {
-		writeTagOutput("0.0.1")
-		os.Exit(0)
+	var prevVersion *semver.Version
+	if config.LatestVersion == "" {
+		prevVersion = latestReleaseVersion
+		// there are no releases on the repo
+		if prevVersion == nil {
+			writeTagOutput("0.0.1")
+			os.Exit(0)
+		}
+	} else {
+		prevVersion, err = semver.NewVersion(config.LatestVersion)
+		if err != nil {
+			fail(fmt.Errorf("--latest-version is not a well-formed semantic version: %w", err))
+		}
+		// Needed for the case where the branch may be versioned to a brand new
+		// version line with no previous releases on it, but we want the first
+		// release to be X.Y.0 (rather than X.Y.1)
+		if prevVersion.GreaterThan(latestReleaseVersion) && prevVersion.Patch() == 0 {
```

Comment:
might need to fix this case

Reply:
change to check if ref is main

Reply:
or check for latest repository release version that is related to latestVersion major.minor.*